controllers/agentcluster_controller.go

```diff
@@ -348,3 +371,37 @@ func (r *AgentClusterReconciler) ensureSecretLabel(ctx context.Context, name, na
 		r.Log.WithError(err).Warn("Failed labeling secret")
 	}
 }
+
+func (r *AgentClusterReconciler) handleDeletion(ctx context.Context, agentCluster *capiproviderv1.AgentCluster) error {
+	if paused := agentCluster.Annotations[clusterv1.PausedAnnotation]; paused == "true" {
+		// unset finalizer, remove owner from owned resources to orphan them and return
+		if err := r.orphanResources(ctx, agentCluster); err != nil {
+			return err
+		}
+	}
+	controllerutil.RemoveFinalizer(agentCluster, agentClusterFinalizer)
```

Comment:
Where does this change get pushed over the API?

Reply:
Good catch! I added a deferred patch to the Reconcile function.